### DISCUSSION TITLE
Make the Audio Bus effects adjustable with Animations and/or Tweens

### DISCUSSION THREAD

**Discussion by HVukman** [godotengine/godot-proposals#6997] [General Discussions]
At the present state you cannot change audio effects while running Godot. See: https://imgur.com/a/HXRix5x
There are no key symbols and you cannot tween the values, since they are not node paths. See this code snippet:

```
# taking the first effect from the audio bus
var effect_0:AudioEffectPitchShift=AudioServer.get_bus_effect(0,0)

# error: argument should be node path but is float
tween.tween_property(effect_0,effect_0.pitch_scale, 0.9,1)
	
#error: argument is null_value
tween.tween_property(effect_0,"pitch_scale", 0.9,1)
```

You can make a new AudioEffect resource and tween the values, but then you have to add the effect to the bus again.

![2023-06-01 14_37_29-vapor tscn - myownvaporwave - Godot Engine](https://github.com/godotengine/godot-proposals/assets/23708144/18bff9fd-9612-48d0-ae39-3398af3a0826)

**Comment by AThousandShips** [comment]
You should be able to do this with a `MethodTweener` no?

The fact that it says that it's a null is a bit strange, are you sure you're getting the right effect? You've added an image of a `AudioEffectDistortion` but your code says `AudioEffectPitchShift`?

**Comment by Calinou** [comment]
You can access the AudioServer singleton from a script to modify various effects at run-time in a script. Use Tween's low-level interpolation methods to interpolate values when calling methods on the AudioEffect instances stored in the audio bus (the values returned by AudioServer).
